src/test/functional/shared/testingSupportApi.ts

```diff
@@ -263,24 +229,91 @@ export const activateUserAccount = async (code: string, token: string) => {
   }
 };
 
-export const createService = async (label: string, description: string, clientId: string, clientSecret: string, redirectUris: string[], onboardingRoles: string[] = []) => {
+export const getTestingServiceClientToken = async () => {
+  const credentials = {
+    'grant_type': 'client_credentials',
+    'client_secret': testConfig.FUNCTIONAL_TEST_SERVICE_CLIENT_SECRET as string,
+    'client_id': testConfig.FUNCTIONAL_TEST_SERVICE_CLIENT_ID as string,
+    'scope': 'profile'
+  };
+  try {
+    return (await axios.post(
+      `${config.get('services.idam.url.api')}/o/token`,
+      new URLSearchParams(credentials)
+    )).data.access_token;
+  } catch (e) {
+    throw new Error(`Failed to getTestingServiceClientToken with ${credentials.client_id}:${credentials.client_secret}, http-status: ${e.response?.status}`);
+  }
+};
+
+export const createRoleFromTestingSupport = async (roleName: string,assignableRoleNames: string[] ) => {
+
+  const accessToken = await getTestingServiceClientToken();
```

Comment:
Do you get a new accessToken for every call? that's going to put all of the roles into different testing sessions which means testing support won't be able to delete the users before attempting to delete the roles. Ideally all the test data (users, roles. services) needs to be created using the same access token so they can all get cleared down together

Reply:
Yes, now changed it